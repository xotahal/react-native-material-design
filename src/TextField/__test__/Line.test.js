import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '../../styles/ThemeProvider.react';

import Line from '../Line.react';

/* eslint-env jest */

[{ type: 'solid', color: 'black' }, { type: 'dotted', color: 'grey' }]
    .forEach(({ type, color }) => {
        it(`renders ${type} ${color} line`, () => {
            const line = renderer
                .create(
                    <ThemeProvider>
                        <Line type={type} color={color} />
                    </ThemeProvider>)
                .toJSON();

            expect(line)
                .toMatchSnapshot();
        });
    });

