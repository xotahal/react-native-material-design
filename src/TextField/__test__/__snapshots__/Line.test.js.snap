// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dotted grey line 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      Array [
        Object {
          "borderColor": "rgba(0, 0, 0, .38)",
          "borderWidth": 1,
          "bottom": 0,
          "left": -1.5,
          "position": "absolute",
          "right": -1.5,
          "top": -2,
        },
      ],
      Object {
        "borderStyle": "dotted",
      },
    ]
  }
/>
`;

exports[`renders solid black line 1`] = `
<View
  pointerEvents="none"
  style={
    Array [
      Array [
        Object {
          "borderColor": "rgba(0, 0, 0, .38)",
          "borderWidth": 1,
          "bottom": 0,
          "left": -1.5,
          "position": "absolute",
          "right": -1.5,
          "top": -2,
        },
      ],
      Object {
        "borderStyle": "solid",
      },
    ]
  }
/>
`;
